rental_pricelist/models/sale.py

```diff
@@ -0,0 +1,320 @@
+                        % line.product_id.name
+                    )
+
+                if line.rental_qty != line.extension_rental_id.rental_qty:
+                    raise ValidationError(
+                        _(
+                            "On the sale order line with rental service %s, "
+                            "you are trying to extend a rental with a rental "
+                            "quantity (%s) that is different from the quantity "
+                            "of the original rental (%s). This is not supported."
+                        )
+                        % (
+                            line.product_id.name,
+                            line.rental_qty,
+                            line.extension_rental_id.rental_qty,
+                        )
+                    )
+            if line.rental_type in ("new_rental", "rental_extension"):
+                if not line.product_id.rented_product_id:
+                    raise ValidationError(
+                        _(
+                            'On the "new rental" sale order line with product '
+                            '"%s", we should have a rental service product!'
+                        )
+                        % (line.product_id.name)
+                    )
+            elif line.sell_rental_id:
+                if line.product_uom_qty != line.sell_rental_id.rental_qty:
+                    raise ValidationError(
+                        _(
+                            "On the sale order line with product %s "
+                            "you are trying to sell a rented product with a "
+                            "quantity (%s) that is different from the rented "
+                            "quantity (%s). This is not supported."
+                        )
+                        % (
+                            line.product_id.name,
+                            line.product_uom_qty,
+                            line.sell_rental_id.rental_qty,
+                        )
+                    )
+
+    @api.onchange("rental_qty", "number_of_time_unit", "product_id")
+    def rental_qty_number_of_days_change(self):
+        if self.product_id.rented_product_id:
+            qty = self.rental_qty * self.number_of_time_unit
+            self.product_uom_qty = qty
+
+    def _get_product_rental_uom_ids(self):
+        self.ensure_one()
+        time_uoms = self._get_time_uom()
+        uom_ids = []
+        if self.display_product_id.rental_of_month:
+            uom_ids.append(time_uoms["month"].id)
+        if self.display_product_id.rental_of_day:
+            uom_ids.append(time_uoms["day"].id)
+        if self.display_product_id.rental_of_hour:
+            uom_ids.append(time_uoms["hour"].id)
+        return uom_ids
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        res = super(SaleOrderLine, self).product_id_change()
+        if self.rental and "domain" in res and "product_uom" in res["domain"]:
```

Comment:
So We should adjust this if clause and add the domain, if res doesn't contain "domain".